pkg/envoy/proxy.go

```diff
@@ -121,12 +122,14 @@ func (p *Proxy) Run(ctx context.Context) error {
 		}
 		return err
 	}
+	metrics.SetGauge([]string{"envoy_connected"}, 1)
 
 	// This goroutine is responsible for waiting on the process (which reaps it
 	// preventing a zombie), triggering cleanup, and notifying the caller that the
 	// process has exited.
 	go func() {
 		err := p.cmd.Wait()
+		metrics.SetGauge([]string{"envoy_connected"}, 0)
```

Comment:
Yeah good feedback. I was really unsure about this metric and since it's basically when we startup/shutdown it seems like it would just be tracking consul dataplane uptime. I'll look into the custom interceptor. I might do it in a subsequent PR though if you are okay with that.

Reply:
Agreed. Adding it here seems like is just capturing `envoy_running` (which could be a useful metric too?).
To Paul's point the xDS proxy [custom handler](https://github.com/hashicorp/consul-dataplane/blob/671d34e9bfdeceb7c9c3bb84f0f9c10e4eef3a78/pkg/consuldp/xds.go#L23) might only be a good indicator that envoy has connected, but not disconnected. Yea, the grpc stream server interceptor seems like a good candidate to detect if envoy is connected/disconnected for xds streaming.

Reply:
So I did a quick implementation of this but there are some problems:
```
func (cdp *ConsulDataplane) streamInterceptor() grpc.StreamServerInterceptor {
	return func(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
	       cdp.logger.Info("In")
		metrics.SetGauge([]string{"envoy_connected"}, 1)
		err := handler(srv, ss) // blocking if connected so envoy not connecting will end up being quite spiky
		metrics.SetGauge([]string{"envoy_connected"}, 0)
		cdp.logger.Info("Out")
		return err
	}
}
```
Since the handler is blocking when connected and the interceptor keeps getting hit over and over when consul is not connected due to retries this will be a very spiky graph and won't ever really sit at 0 for very long. 

It might be better to use a counter that decreases each time we the interceptor handles it so we see envoy_connected is slowly but steadily increasing (healthy) or starting to drop down when it's not connected or failing. Thoughts?

Reply:
Okay I figured a way around this by using a wrapper around the grpc.ServerStream in 1e75aab727f484db9c7d6ba9aa7572043b3e4de6